bindings/nodejs/src/functions/oic-server.cc

```diff
@@ -0,0 +1,262 @@
+    Local<Object> outputPayload = Nan::New<Object>();
+    Local<Value> arguments[3] = {
+        js_sol_network_link_addr(cliaddr),
+        js_sol_oic_map_reader(input),
+        outputPayload
+    };
+    Nan::Callback *callback = info->handlers[method];
+    if (callback) {
+        Local<Value> jsReturnValue = callback->Call(3, arguments);
+        VALIDATE_CALLBACK_RETURN_VALUE_TYPE(jsReturnValue, IsUint32,
+            "entity handler", returnValue);
+        returnValue = (sol_coap_responsecode_t)
+            Nan::To<uint32_t>(jsReturnValue).FromJust();
+
+        if (!c_sol_oic_map_writer(outputPayload, output)) {
+            Nan::ThrowError("entity handler: Failed to encode output payload");
+        }
+    }
+    return returnValue;
+}
+
+static sol_coap_responsecode_t defaultGet(ENTITY_HANDLER_SIGNATURE) {
+    return entityHandler(cliaddr, data, input, output, OIC_SERVER_GET);
+}
+
+static sol_coap_responsecode_t defaultPut(ENTITY_HANDLER_SIGNATURE) {
+    return entityHandler(cliaddr, data, input, output, OIC_SERVER_PUT);
+}
+
+static sol_coap_responsecode_t defaultPost(ENTITY_HANDLER_SIGNATURE) {
+    return entityHandler(cliaddr, data, input, output, OIC_SERVER_POST);
+}
+
+static sol_coap_responsecode_t defaultDel(ENTITY_HANDLER_SIGNATURE) {
+    return entityHandler(cliaddr, data, input, output, OIC_SERVER_DEL);
+}
+
+#define ASSIGN_STR_SLICE_MEMBER_FROM_PROPERTY(to, from, message, member) \
+    do { \
+        to.member.data = strdup((const char *)*String::Utf8Value( \
+            Nan::Get(from, Nan::New(#member).ToLocalChecked()) \
+                .ToLocalChecked())); \
+        if (!to.member.data) { \
+            message = "Failed to allocate memory for " #member; \
+            goto member##_failed; \
+        } \
+        to.member.len = strlen(to.member.data); \
+    } while(0)
+
+static bool c_sol_oic_resource_type(Local<Object> js,
+    struct sol_oic_resource_type *definition) {
+    const char *error = 0;
+    struct sol_oic_resource_type local = {
+#ifdef SOL_OIC_RESOURCE_TYPE_API_VERSION
+        .api_version = SOL_OIC_RESOURCE_TYPE_API_VERSION,
+#endif /* def SOL_OIC_RESOURCE_TYPE_API_VERSION */
+        .resource_type = {0, 0},
+        .interface = {0, 0},
+        .path = {0, 0},
+        .get = { .handle = defaultGet },
+        .put = { .handle = defaultPut },
+        .post = { .handle = defaultPost },
+        .del = { .handle = defaultDel }
+    };
+
+    ASSIGN_STR_SLICE_MEMBER_FROM_PROPERTY(local, js, error, resource_type);
+    ASSIGN_STR_SLICE_MEMBER_FROM_PROPERTY(local, js, error, interface);
+    ASSIGN_STR_SLICE_MEMBER_FROM_PROPERTY(local, js, error, path);
+
+    *definition = local;
+    return true;
+
+path_failed:
+    free((void *)(local.interface.data));
+interface_failed:
+    free((void *)(local.resource_type.data));
+resource_type_failed:
+    Nan::ThrowError(error);
+    return false;
+}
+
+NAN_METHOD(bind_sol_oic_server_add_resource) {
```

Comment:
```NAN_METHOD(x)``` roughly translates as ```void x(something info)```, where ```info``` has these:
* ```info[0], info[1], ... info[n]``` are JS values (```v8::Local<v8::Value>```) that were passed to the binding on the JS side
*```info.GetReturnValue()``` gives you a slot into which you can place the ```v8::Local<v8::Value>``` which should become the return value of the binding on the JS side.